frontend/src/selectors/exerciseSelectors.js

```diff
@@ -18,20 +18,25 @@ export const getExerciseTimer = createSelector(
 );
 
 export const getThreads = createSelector(exerciseSelector, exercise =>
-  exercise.threads.filter(
-    thread =>
-      exercise.emailRevealTimes.filter(time => time.emailId === thread.id)[0]
-        .revealTime <= exercise.timer
-  )
+  exercise.threads
+    .filter(thread => thread.revealTime <= exercise.timer)
```

Comment:
Bloody hell that's confusing! Bet they didn't think about this scenario when implementing arrow functions!!